server/src/service.rs

```diff
@@ -410,4 +416,50 @@ impl Service {
 
         n_sessions
     }
+
+    pub async fn watch_clients(&self) -> Result<(), Error> {
+        // wip: The idea is store client information when a client calls a method.
+        // Watch for NameOwnerChanged signal from the same client.
+        // Clean up session and resources based on that.
+        let connect_rule = zbus::MatchRule::builder()
+            .msg_type(zbus::message::Type::MethodCall)
+            .path_namespace("/org/freedesktop/secrets")?
+            .build();
```

Comment:
This seems like too broad, a client has to start a session somewhere first no? so that is where i would start the tracking :-)

Reply:
Thanks for the prompt review. Yeah, I agree :)